ixmp/core.py

```diff
@@ -583,9 +592,20 @@ def timeseries(self, region=None, variable=None, unit=None, year=None,
         df['model'] = self.model
         df['scenario'] = self.scenario
 
+        # drop `subannual` column if not requested (False) or required ('auto')
+        if subannual is not True:
+            has_subannual = any(df['subannual'].unique() != ['Year'])
+            if subannual is False and has_subannual:
+                msg = ("timeseries data has subannual values, ",
+                       "use `subannual=True or 'auto'`")
+                raise ValueError(msg)
+            if not has_subannual:
+                df.drop('subannual', axis=1, inplace=True)
+
         if iamc:
             # Convert to wide format
-            index = IAMC_IDX + ['subannual']
+            index = IAMC_IDX + ['subannual'] if 'subannual' in df.columns \
```

Comment:
nit: a bit hard to read as it needs to think/recall [operator precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence). I would do it more explicit.

Reply:
changer for a more intuitive implementation